main/src/library/File.flix

```diff
@@ -309,19 +309,29 @@ namespace File {
             import new java.lang.String(Array[Int8, Static], ##java.nio.charset.Charset): String & Impure as newString;
 
             let javaFile = newFile(f);
-            let stream = newInputStream(javaFile);
-            discard skip(stream, opts.offset);
-            let bytes = [0i8; opts.count];
-            discard read(stream, bytes, 0, opts.count);
-            let charSet = forName(opts.charSet);
-            let returnString = newString(bytes, charSet);
-            Ok(returnString)
+            def loop(i) = {
+                if (0 <= i) {
+                    let stream = newInputStream(javaFile);
+                    let skipped = skip(stream, opts.offset);
+                    if (skipped == opts.offset) {
+                        let bytes = [0i8; opts.count];
+                        discard read(stream, bytes, 0, opts.count);
+                        let charSet = forName(opts.charSet);
+                        let returnString = newString(bytes, charSet);
+                        Ok(returnString)
+                    } else {
+                        loop(i - 1)
+                    }
+                } else {
+                    Err("An error occurred when reading the file: '${f}'.")
+                }
+            };
+            loop(opts.count / 1000)
```

Comment:
I am not sure how many times we want to try to skip the correct number of bytes

Reply:
I think it should use `opts.offset` instead of `opts.count` since offset is the argument to skip.

I have no idea what number to divide by but `opts.offset/1000` will be a retry for every kilobyte which is.. maybe a bit high? maybe 10.000 or 100.000.